tools/patch/tensorflow/cc/saved_model/loader.cc

```diff
@@ -1,32 +1,36 @@
---- tensorflow/cc/saved_model/loader.cc	2018-10-19 10:25:04.454294410 -0700
-+++ ../tensorrtserver/tools/patch/tensorflow/cc/saved_model/loader.cc	2018-11-02 11:47:57.254526547 -0700
-@@ -19,6 +19,7 @@
+diff --git a/tensorflow/cc/saved_model/loader.cc b/tensorflow/cc/saved_model/loader.cc
+index 85d3dd01fa..b385f8b857 100644
+--- a/tensorflow/cc/saved_model/loader.cc
++++ b/tensorflow/cc/saved_model/loader.cc
+@@ -19,6 +19,7 @@ limitations under the License.
  
  #include "tensorflow/cc/saved_model/constants.h"
  #include "tensorflow/cc/saved_model/reader.h"
 +#include "tensorflow/core/graph/default_device.h"
  #include "tensorflow/core/lib/io/path.h"
  #include "tensorflow/core/lib/monitoring/counter.h"
  #include "tensorflow/core/lib/strings/str_util.h"
-@@ -216,8 +217,20 @@
+@@ -244,9 +245,22 @@ Status LoadSavedModelInternal(const SessionOptions& session_options,
+                               SavedModelBundle* const bundle) {
    TF_RETURN_IF_ERROR(ReadMetaGraphDefFromSavedModel(export_dir, tags,
                                                      &bundle->meta_graph_def));
- 
-+  // If visible_device_list starts with a '/' then it is being used to
++  
++  // If allocator starts with a '/' then it is being used to
 +  // communicate the CPU/GPU that the graph runs on. This isn't
 +  // foolproof since individual operations in the graph could specify
 +  // a specific run location. [DLIS-43]
 +  SessionOptions lsession_options = session_options;
-+  const std::string& vdl =
-+    lsession_options.config.gpu_options().visible_device_list();
-+  if (!vdl.empty() && (vdl[0] == '/')) {
-+    graph::SetDefaultDevice(vdl, bundle->meta_graph_def.mutable_graph_def());
-+    lsession_options.config.mutable_gpu_options()->clear_visible_device_list();
++  const std::string& alloc_type =
++    lsession_options.config.gpu_options().allocator_type();
++  if (!alloc_type.empty() && (alloc_type[0] == '/')) {
++    graph::SetDefaultDevice(alloc_type, bundle->meta_graph_def.mutable_graph_def());
++    lsession_options.config.mutable_gpu_options()->clear_allocator_type();
```

Comment:
@tanmayv25 Is this related to what you are trying to do with TF-TRT models?

Reply:
Tanmay seems to also be working on this code. This may lead to some merge issues later.

Reply:
It is fine as long as you guys are working on different things. I just worried that he was working on something that has been implemented.